### PR TITLE
[oidc] find/create User account

### LINKED ISSUE #14954: [sso/oidc] Add `/session` endpoint to `server`
part of Epic: Single sign-on (SSO) #7761

In an [internal discussion](https://www.notion.so/gitpod/OpenID-Connect-SSO-integration-ce0f915f1e2d4329bc2a00a2a6f9d7a7#39fd8f955e7144cbaca6e221c5be0613) we've concluded that it would be fine to add this endpoint intermediately, which would allow us to start out without adding session management to `iam` component.  

* To be called by `iam` services internally, thus must not be exposed.
* The purpose is to create a (temporary) shortcut and use server's implementation to touch a session for a user after the OIDC flow results (id_token, token, user profile) are fetched.
* Let's use a shared secret to authorize the call.

### PR DESCRIPTION
**Pull request by AlexTugarev**
During the OIDC flow, the `iam` component does a `POST` request to the server internally which should ensure a HTTP session is created to proceed with. This PR comes with the implementation of an account look-up.

To simplify things, I decided to map `authProviderId` to `issuerID` and `authId` to `subject`. There is no complete mapping of auth providers to OIDC clients included, as it might be not required and potentially wrong.

## Related Issue(s)

Based on #15629
Part of #7761
Fixes #14954

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/src/iam/iam-session-app.spec.ts
Comment (easyCZ) [inline components/server/src/iam/iam-session-app.spec.ts:115]: Why is the body being converted to a string, when we're only check for the status code to be 200?

File: components/server/src/iam/iam-session-app.ts
Comment (easyCZ) [inline components/server/src/iam/iam-session-app.ts:47]: Invalid argument error?

File: components/server/src/iam/iam-session-app.ts
Comment (easyCZ) [inline components/server/src/iam/iam-session-app.ts:49]: Do we need to validate the claims here?

File: components/server/src/iam/iam-session-app.spec.ts
Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.spec.ts:115]: that prints potential errors if status code is not 200.

File: components/server/src/iam/iam-session-app.ts
Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.ts:49]: sounds good, at least for those which are currently in use.

File: components/server/src/iam/iam-session-app.ts
Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.ts:47]: that will be wrapped into a code 500 in the handler.

Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.ts:47]: that's true.

Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.ts:48]: done

File: components/server/src/iam/iam-session-app.ts
Comment (easyCZ) [inline components/server/src/iam/iam-session-app.ts:47]: This validation check just checks that the fields are present, but doesn't validate the values. Do we also need to validate these?

File: components/server/src/iam/iam-session-app.ts
Comment (easyCZ) [reply components/server/src/iam/iam-session-app.ts:50]: I'm not seeing the validation of the values, what am I missing?

File: components/server/src/iam/iam-session-app.ts
Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.ts:47]: Validation against what exactly? 
That's the ID token from an OIDC provider, which is validated in and pass on by the `iam` [internally](https://github.com/gitpod-io/gitpod/blob/243207eb85e1ac9ada023fc6ba3a4e3118d49d5e/components/iam/pkg/oidc/router.go#L89-L102).

Comment (easyCZ) [reply components/server/src/iam/iam-session-app.ts:47]: Sure. But every service should perform at least basic sanity checks on the fields. If nothing else, we should check that the fields are not empty. This acts as a safeguard if we get it wrong on the other service

Comment (geropl) [reply components/server/src/iam/iam-session-app.ts:47]: I mean, we need to rely on external services to adhere to their own contracts.
But I get the idea that - because this is the external boundary of "our" system - we want to be super-clear that we provide good, stable guarantees to all other "internal" components.

**Review by geropl — APPROVED** [review]
Code itself LGTM. Approving to unblock,

/hold in case you want to resolve [this discussion](https://github.com/gitpod-io/gitpod/pull/15631/files#r1064742206) in this PR

File: components/server/src/iam/iam-session-app.ts
Comment (AlexTugarev) [reply components/server/src/iam/iam-session-app.ts:47]: Done [here](https://github.com/gitpod-io/gitpod/compare/a4035031234857f90d9f6f96171aafe10ae01a94..0d5c0d6dc02e5d4bca61482fe7baf3dd5afd587e)

**Pull request merged**
